src/borg/testsuite/archiver.py

```diff
@@ -2841,6 +2841,10 @@ def test_get_args():
                              'borg init --encryption=repokey /')
     assert args.func == archiver.do_serve
 
+    args = archiver.get_args(['borg', 'serve', ],
+                             'BORG_HOSTNAME_IS_UNIQUE=yes borg init --encryption=repokey /')
+    assert args.func == archiver.do_serve
```

Comment:
did you intentionally test "trying to cheat" **and** env var removal?

maybe put a comment above it, like in the test cases above?